sdk/anomalydetector/ai-anomaly-detector-rest/samples-dev/sample_detect_change_point.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Demonstrates how to detect change points on entire series.
+ *
+ * @summary detects change points.
+ */
+
+import AnomalyDetector, {
+  ChangePointDetectResponseOutput,
+  DetectUnivariateChangePointParameters,
+  isUnexpected,
+  TimeSeriesPoint,
+} from "@azure-rest/ai-anomaly-detector";
+import { AzureKeyCredential } from "@azure/core-auth";
+
+import { parse } from "csv-parse/sync";
+import * as fs from "fs";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// You will need to set this environment variables or edit the following values
+
+const apiKey = process.env["ANOMALY_DETECTOR_API_KEY"] || "";
+const endpoint = process.env["ANOMALY_DETECTOR_ENDPOINT"] || "";
+const apiVersion = "v1.1";
+const timeSeriesDataPath = "./samples-dev/example-data/request-data.csv";
+
+function read_series_from_file(path: string): Array<TimeSeriesPoint> {
+  let result = Array<TimeSeriesPoint>();
+  let input = fs.readFileSync(path).toString();
+  let parsed = parse(input, { skip_empty_lines: true });
+  parsed.forEach(function (e: Array<string>) {
+    result.push({ timestamp: new Date(e[0]), value: Number(e[1]) });
+  });
+  return result;
+}
+
+export async function main() {
+  const credential = new AzureKeyCredential(apiKey);
+  const client = AnomalyDetector(endpoint, apiVersion, credential);
+  const options: DetectUnivariateChangePointParameters = {
+    body: {
+      granularity: "daily",
+      series: read_series_from_file(timeSeriesDataPath),
+    },
+    headers: { "Content-Type": "application/json" },
+  };
+  const result = await client.path("/timeseries/changepoint/detect").post(options);
+  if (isUnexpected(result)) {
+    throw result;
+  }
+
+  if (
+    (result.body as ChangePointDetectResponseOutput).isChangePoint!.some(function (changePoint) {
```

Comment:
We should remove the explicit cast. isUnexpected should narrow down to the success types. If there are more than one success types you should narrow it down further with `if(result.status === "200") {} else {}` 

Similarly, we shouldn't have `isChangePoint!` you should actually null check. If the service never returns undefined, then the swagger needs to be updated to make this property required.

TBH it feels a bit weird that `isChangePoint` is a collection, I would have expected it to be a boolean